app/src/main/java/org/oppia/android/app/options/OptionsActivity.kt

```diff
@@ -115,23 +119,24 @@ class OptionsActivity :
       profileId!!
     )
     title = resourceHandler.getStringInLocale(R.string.menu_options)
-  }
 
-  override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
-    super.onActivityResult(requestCode, resultCode, data)
-    checkNotNull(data) {
-      "Expected data to be passed as an activity result for request: $requestCode."
-    }
-    when (requestCode) {
-      REQUEST_CODE_TEXT_SIZE -> {
-        val textSizeResults = data.getProtoExtra(
+    readingTextSizeLauncher = registerForActivityResult(
+      ActivityResultContracts.StartActivityForResult()
+    ) { result ->
+      if (result.resultCode == RESULT_OK && result.data != null) {
+        val textSizeResults = result.data!!.getProtoExtra(
```

Comment:
Here and elsewhere, suggest not using `!!`.
```suggestion
   readingTextSizeLauncher = registerForActivityResult(
  ActivityResultContracts.StartActivityForResult()
) { result ->
  if (result.resultCode == RESULT_OK && result.data != null) {
    val textSizeResults = result.data?.getProtoExtra(
      MESSAGE_READING_TEXT_SIZE_RESULTS_KEY,
      ReadingTextSizeActivityResultBundle.getDefaultInstance()
    )
    if (textSizeResults != null) {
      optionActivityPresenter.updateReadingTextSize(textSizeResults.selectedReadingTextSize)
    }
  }
}
```

Reply:
Done.